packages/devtools-client-adapters/src/chrome.js

```diff
@@ -32,85 +32,80 @@ function createTabs(tabs, {type, clientType} = {}) {
     });
 }
 
-function connectClient() {
-  const deferred = defer();
+window.criRequest = function(options, callback) {
+  const { host, port, path } = options;
+  const url = `http://${host}:${port}${path}`
+
+  networkRequest(url)
+    .then(res => callback(null, res.content))
+    .catch(err => callback(err))
+}
 
+function connectClient() {
   if(!getValue("chrome.debug")) {
     return Promise.resolve(createTabs([]))
   }
 
-  const port = getValue("chrome.port");
-  const host = getValue("chrome.host");
-  const url = `http://${host}:${port}/json/list`;
-
-  networkRequest(url).then(res => {
-    deferred.resolve(createTabs(
-      JSON.parse(res.content),
-      {clientType: "chrome", type: "page"}
-    ))
-  }).catch(err => deferred.reject());
-
-  return deferred.promise;
+  return CDP.List({
+    port: getValue("chrome.port"),
+    host: getValue("chrome.host")
+  })
+    .then(tabs => createTabs(tabs, {
+      clientType: "chrome", type: "page"
+    }))
 }
 
 
 function connectNodeClient() {
-  const deferred = defer();
-
   if(!getValue("node.debug")) {
     return Promise.resolve(createTabs([]))
   }
 
-  const host = getValue("node.host");
-  const port = getValue("node.port");
-  const url = `http://${host}:${port}/json/list`;
-
-  networkRequest(url).then(res => {
-    deferred.resolve(createTabs(
-      JSON.parse(res.content),
-      {clientType: "node", type: "node"}
-    ))
-  }).catch(err => {
-    console.log(err);
-    deferred.reject();
-  });
-
-  return deferred.promise;
+  return CDP.List({
+    port: getValue("node.port"),
+    host: getValue("node.host")
+  })
+    .then(tabs => createTabs(tabs, {
+      clientType: "node", type: "node"
+    }))
 }
 
 function connectTab(tab) {
-  return connect(tab.webSocketDebuggerUrl, {type: "browser"})
+  return CDP(tab.webSocketDebuggerUrl)
```

Comment:
If I'm reading correctly the code at 
- https://github.com/cyrus-and/chrome-remote-interface/blob/master/index.js
- https://github.com/cyrus-and/chrome-remote-interface/blob/master/lib/chrome.js

CDP expects options containing host/port/protocol etc and not a string representing a URL ?